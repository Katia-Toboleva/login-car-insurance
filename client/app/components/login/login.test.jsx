import React from 'react';
import { mount } from 'enzyme';
import Login from './login';

const getUnit = (props) => (
  mount(<Login {...props} />)
);

describe('Login', () => {
  describe('render', () => {
    it('should render correctly with default values(pending status)', () => {
      const wrapper = getUnit();

      expect(wrapper).toMatchSnapshot();
    });

    it('should render correctly with loginStatus rejected', () => {
      const wrapper = getUnit({ loginStatus: 'rejected' });

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('handleLoginSubmit', () => {
    it('should call onLoginSubmit', () => {
      const cb = jest.fn();
      const wrapper = getUnit({ onLoginSubmit: cb });
      const form = wrapper.find('form').at(0);
      form.simulate('submit',
        { preventDefault () {} }
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});
