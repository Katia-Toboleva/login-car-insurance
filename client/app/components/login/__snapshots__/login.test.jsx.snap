// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login render should render correctly with default values(pending status) 1`] = `
<div
  className="login__wrapper"
>
  <div
    className="login"
  >
    <div
      className="login__title"
    >
      <Text
        center={true}
        color="black"
        size="large"
        text="sign in"
        transform="capitalize"
        weight="bold"
      />
    </div>
    <div
      className="login__body"
    >
      <Input
        onChange={[Function]}
        onKeyUp={[Function]}
        placeholder="user name"
        type="text"
        value=""
      />
      <div
        className="login__password"
      >
        <Input
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="password"
          type="password"
          value=""
        />
        <div
          className="login__password__icon"
        >
          <Icon
            color="grey"
            icon="closedEye"
            onClick={[Function]}
            size="medium"
          />
        </div>
      </div>
    </div>
    <Spinner
      absolute={false}
    />
    <div
      className="login__controls"
    >
      <Button
        active={false}
        onClick={[Function]}
        size="medium"
        text="sign in"
        type="login"
      />
    </div>
  </div>
</div>
`;

exports[`Login render should render correctly with loginStatus rejected 1`] = `
<div
  className="login__wrapper"
>
  <div
    className="login"
  >
    <div
      className="login__title"
    >
      <Text
        center={true}
        color="black"
        size="large"
        text="sign in"
        transform="capitalize"
        weight="bold"
      />
    </div>
    <div
      className="login__body"
    >
      <Input
        onChange={[Function]}
        onKeyUp={[Function]}
        placeholder="user name"
        type="text"
        value=""
      />
      <div
        className="login__password"
      >
        <Input
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="password"
          type="password"
          value=""
        />
        <div
          className="login__password__icon"
        >
          <Icon
            color="grey"
            icon="closedEye"
            onClick={[Function]}
            size="medium"
          />
        </div>
      </div>
    </div>
    <div>
      Oops! Wrong Username or Password
    </div>
    <div
      className="login__controls"
    >
      <Button
        active={false}
        onClick={[Function]}
        size="medium"
        text="sign in"
        type="login"
      />
    </div>
  </div>
</div>
`;
